From patchwork Mon Aug 26 11:06:15 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: Pierre-Jean Texier <pjtexier@koncepto.io>
X-Patchwork-Id: 1153103
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org; dmarc=fail (p=quarantine dis=none)
 header.from=koncepto.io
Authentication-Results: ozlabs.org;
 dkim=fail reason="signature verification failed" (2048-bit key;
 unprotected) header.d=koncepto.io header.i=@koncepto.io
 header.b="oiyEB9Yh"; dkim-atps=neutral
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 46H8NK1LPKz9sDB
 for <incoming@patchwork.ozlabs.org>;
 Mon, 26 Aug 2019 21:07:13 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id CAF1DC21F7D; Mon, 26 Aug 2019 11:06:58 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=SPF_HELO_PASS, T_DKIM_INVALID
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 0AA54C21FCA;
 Mon, 26 Aug 2019 11:06:52 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 27147C21FB5; Mon, 26 Aug 2019 11:06:47 +0000 (UTC)
Received: from koncepto.io (koncepto.io [195.154.119.111])
 by lists.denx.de (Postfix) with ESMTPS id A73A1C21F9A
 for <u-boot@lists.denx.de>; Mon, 26 Aug 2019 11:06:44 +0000 (UTC)
Received: from menoah.home (lfbn-1-8923-205.w193-250.abo.wanadoo.fr
 [193.250.65.205])
 by koncepto.io (Postfix) with ESMTPSA id 0C8AF602CF;
 Mon, 26 Aug 2019 13:06:44 +0200 (CEST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=koncepto.io;
 s=default; t=1566817604;
 bh=V4r1z9HD8aVGoyELWRC99LX8pZowixLpYqFHkVfjVik=;
 h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
 b=oiyEB9YhFL8jh+kux8tsm0WhBM8b/ijFR0262b6Oe55f6YePlv9wHb6b1pEW1Fs62
 gNimouSjMWLCAW2wZquTDRjoQoRVCykS9rPbigionFMazyPmt1F0vDBpjoN75MDJl9
 yx8GGplCS3yU5fHLkfd1SKxnta0u7sLu7EnjfagoI6XuV53Ix6DkJAvkvPfgUhl1hD
 2Cz7oBfJafbblAdKb019EWk6TTrW86ASL0aHwKF/R3hXfDjwLN2gzhD0pckzmEklxR
 Z+9c649IYaa9B3K3PAdhRoQ5lptYyVqLNrAffsabjbXQzrxyBcGEg5hRnleeede86M
 4HoWxUCDgE5EQ==
From: Pierre-Jean Texier <pjtexier@koncepto.io>
To: u-boot@lists.denx.de
Date: Mon, 26 Aug 2019 13:06:15 +0200
Message-Id: <1566817578-16750-2-git-send-email-pjtexier@koncepto.io>
X-Mailer: git-send-email 2.7.4
In-Reply-To: <1566817578-16750-1-git-send-email-pjtexier@koncepto.io>
References: <1566817578-16750-1-git-send-email-pjtexier@koncepto.io>
MIME-Version: 1.0
Cc: bjorn@haxx.se, michal.simek@xilinx.com, joe.hershberger@ni.com,
 philip@philipmolloy.com
Subject: [U-Boot] [PATCH v2 1/4] fw_env: remove duplicated definitions
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

Since commit d3716dd ("env: Rename the redundancy flags"), the
definitions of ENV_REDUND_OBSOLETE & ENV_REDUND_ACTIVE was moved
to env.h.

Fixes:

tools/env/fw_env.c:122:22: error: ‘ENV_REDUND_ACTIVE’ redeclared as different kind of symbol
 static unsigned char ENV_REDUND_ACTIVE = 1;
                      ^~~~~~~~~~~~~~~~~
In file included from tools/env/fw_env.c:13:
include/env.h:63:2: note: previous definition of ‘ENV_REDUND_ACTIVE’ was here
  ENV_REDUND_ACTIVE = 1,
  ^~~~~~~~~~~~~~~~~
tools/env/fw_env.c:127:22: error: ‘ENV_REDUND_OBSOLETE’ redeclared as different kind of symbol
 static unsigned char ENV_REDUND_OBSOLETE;
                      ^~~~~~~~~~~~~~~~~~~
In file included from tools/env/fw_env.c:13:
include/env.h:62:2: note: previous definition of ‘ENV_REDUND_OBSOLETE’ was here
  ENV_REDUND_OBSOLETE = 0,

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
Tested-by: Joris Offouga <offougajoris@gmail.com>
Tested-by: Heiko Schocher <hs@denx.de>
---
v1 -> v2:
 - None
 
 tools/env/fw_env.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/tools/env/fw_env.c b/tools/env/fw_env.c
index 95c9984..876bf2b 100644
--- a/tools/env/fw_env.c
+++ b/tools/env/fw_env.c
@@ -119,13 +119,6 @@ static struct environment environment = {
 
 static int have_redund_env;
 
-static unsigned char ENV_REDUND_ACTIVE = 1;
-/*
- * ENV_REDUND_OBSOLETE must be 0 to efficiently set it on NOR flash without
- * erasing
- */
-static unsigned char ENV_REDUND_OBSOLETE;
-
 #define DEFAULT_ENV_INSTANCE_STATIC
 #include <env_default.h>
 

From patchwork Mon Aug 26 11:06:16 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: Pierre-Jean Texier <pjtexier@koncepto.io>
X-Patchwork-Id: 1153106
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org; dmarc=fail (p=quarantine dis=none)
 header.from=koncepto.io
Authentication-Results: ozlabs.org;
 dkim=fail reason="signature verification failed" (2048-bit key;
 unprotected) header.d=koncepto.io header.i=@koncepto.io
 header.b="c5/r1UEA"; dkim-atps=neutral
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 46H8Q906lcz9sBF
 for <incoming@patchwork.ozlabs.org>;
 Mon, 26 Aug 2019 21:08:48 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id AEEDAC2212D; Mon, 26 Aug 2019 11:07:36 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=SPF_HELO_PASS, T_DKIM_INVALID
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 858A4C2214E;
 Mon, 26 Aug 2019 11:07:01 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id BFF15C2216A; Mon, 26 Aug 2019 11:06:48 +0000 (UTC)
Received: from koncepto.io (koncepto.io [195.154.119.111])
 by lists.denx.de (Postfix) with ESMTPS id 87FBDC21FCA
 for <u-boot@lists.denx.de>; Mon, 26 Aug 2019 11:06:45 +0000 (UTC)
Received: from menoah.home (lfbn-1-8923-205.w193-250.abo.wanadoo.fr
 [193.250.65.205])
 by koncepto.io (Postfix) with ESMTPSA id EB51860367;
 Mon, 26 Aug 2019 13:06:44 +0200 (CEST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=koncepto.io;
 s=default; t=1566817605;
 bh=/BzW5h1tpm92Axd1WgUReyBQsFQF3cTqd1CNLOzC1ko=;
 h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
 b=c5/r1UEA9buu4jObmZtyFpyrpiS53PaMZNrL4rMiyrbgACz9Mkdy2Z55vsSjmATqG
 7uTbEVjsPuqi/ECX2+8I9ozMGh72h4vosbfoak2jzAwHkC4ostork7ltSS2ahy7ZuO
 CN0oXYk6/WA5mipEaQb6aaipBMd65Hm7hfN365ZHk5a8RBJD805QekonFLGUahFIP6
 J7bu7RkeQeCV5n3lqIPfYddP73Lfx4hNs5plSHs+xDT69dyfW7xBGp0qLEljRfw9VS
 WD/O/BhsUWgQVVbQdludCJaYQgrkLYRUOjdiAA8zMCFYX7TnQT5wvzHDSQiZ0w+RDW
 ClVzkMBJWfW4Q==
From: Pierre-Jean Texier <pjtexier@koncepto.io>
To: u-boot@lists.denx.de
Date: Mon, 26 Aug 2019 13:06:16 +0200
Message-Id: <1566817578-16750-3-git-send-email-pjtexier@koncepto.io>
X-Mailer: git-send-email 2.7.4
In-Reply-To: <1566817578-16750-1-git-send-email-pjtexier@koncepto.io>
References: <1566817578-16750-1-git-send-email-pjtexier@koncepto.io>
MIME-Version: 1.0
Cc: bjorn@haxx.se, michal.simek@xilinx.com, joe.hershberger@ni.com,
 philip@philipmolloy.com
Subject: [U-Boot] [PATCH v2 2/4] fw_env: fix build error
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

The following error appears:

tools/env/fw_env.c:1149:25: error: lvalue required as unary ‘&’ operand
  rc = write(fd, &ENV_REDUND_OBSOLETE, sizeof(ENV_REDUND_OBSOLETE));

Fixes: d3716dd ("env: Rename the redundancy flags")

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
Tested-by: Joris Offouga <offougajoris@gmail.com>
Tested-by: Heiko Schocher <hs@denx.de>
Suggested-by: Heiko Schocher <hs@denx.de>
---
v1 -> v2:
 - fix fw_setenv with flag_obsolete (reported by Heiko Schocher)

 tools/env/fw_env.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tools/env/fw_env.c b/tools/env/fw_env.c
index 876bf2b..e2801f5 100644
--- a/tools/env/fw_env.c
+++ b/tools/env/fw_env.c
@@ -1135,6 +1135,7 @@ static int flash_flag_obsolete(int dev, int fd, off_t offset)
 {
 	int rc;
 	struct erase_info_user erase;
+	char tmp = ENV_REDUND_OBSOLETE;
 
 	erase.start = DEVOFFSET(dev);
 	erase.length = DEVESIZE(dev);
@@ -1146,7 +1147,7 @@ static int flash_flag_obsolete(int dev, int fd, off_t offset)
 		return rc;
 	}
 	ioctl(fd, MEMUNLOCK, &erase);
-	rc = write(fd, &ENV_REDUND_OBSOLETE, sizeof(ENV_REDUND_OBSOLETE));
+	rc = write(fd, &tmp, sizeof(tmp));
 	ioctl(fd, MEMLOCK, &erase);
 	if (rc < 0)
 		perror("Could not set obsolete flag");

From patchwork Mon Aug 26 11:06:17 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: Pierre-Jean Texier <pjtexier@koncepto.io>
X-Patchwork-Id: 1153105
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org; dmarc=fail (p=quarantine dis=none)
 header.from=koncepto.io
Authentication-Results: ozlabs.org;
 dkim=fail reason="signature verification failed" (2048-bit key;
 unprotected) header.d=koncepto.io header.i=@koncepto.io
 header.b="N3SBJxTS"; dkim-atps=neutral
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 46H8PF6lSpz9sBF
 for <incoming@patchwork.ozlabs.org>;
 Mon, 26 Aug 2019 21:08:01 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id 98EDDC2213B; Mon, 26 Aug 2019 11:07:23 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=SPF_HELO_PASS, T_DKIM_INVALID
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 118FFC21F9C;
 Mon, 26 Aug 2019 11:06:59 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 84A9AC220FB; Mon, 26 Aug 2019 11:06:49 +0000 (UTC)
Received: from koncepto.io (koncepto.io [195.154.119.111])
 by lists.denx.de (Postfix) with ESMTPS id 45D40C22127
 for <u-boot@lists.denx.de>; Mon, 26 Aug 2019 11:06:46 +0000 (UTC)
Received: from menoah.home (lfbn-1-8923-205.w193-250.abo.wanadoo.fr
 [193.250.65.205])
 by koncepto.io (Postfix) with ESMTPSA id 99A5960378;
 Mon, 26 Aug 2019 13:06:45 +0200 (CEST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=koncepto.io;
 s=default; t=1566817605;
 bh=dWwf9DG0EIYj1c3i6TZ9GcIjpKB4dt6vWg1Dt8wKqzw=;
 h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
 b=N3SBJxTSaw0Cc/Af4mtNxZoxnO00iPmrwKFdSpv7Ivcye0Fvia3QcbbWuikUWWsvC
 tJmF5PWGKgVlihn9XPEGhMKbhKWPGiBu/s2hlexR6DoJJxDf/8NmNTh8lpWjm4OvRM
 PL7ZRhxs8bIN8CO+pKk+FwZswp8J/ox8Y65QcrmlG7eyLivK+/D87dI0GPVjeu68Am
 5LC+rYJKFB12qalI8b+QJHWZFyhx4yA6crHbz3ynfXYMFywXLz99bvj+3hiyadKuVA
 HvY54Esmj1mVtaO9YeWOA0UseI6tI7NnFpFsDhXOHGjROrGPlGH+ZutRy00m1sfcZH
 hb5LdaY/5drGg==
From: Pierre-Jean Texier <pjtexier@koncepto.io>
To: u-boot@lists.denx.de
Date: Mon, 26 Aug 2019 13:06:17 +0200
Message-Id: <1566817578-16750-4-git-send-email-pjtexier@koncepto.io>
X-Mailer: git-send-email 2.7.4
In-Reply-To: <1566817578-16750-1-git-send-email-pjtexier@koncepto.io>
References: <1566817578-16750-1-git-send-email-pjtexier@koncepto.io>
MIME-Version: 1.0
Cc: bjorn@haxx.se, michal.simek@xilinx.com, joe.hershberger@ni.com,
 philip@philipmolloy.com
Subject: [U-Boot] [PATCH v2 3/4] env: add missing <compiler.h> header file
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

Since commit af95f20 ("env: Create a new file for environment functions"),
a new header file exists.

So, this commit add a missing header file.

Fixes:

include/env.h:158:1: error: unknown type name ‘ulong’; did you mean ‘long’?
 ulong env_get_ulong(const char *name, int base, ulong default_val);
 ^~~~~
 long
include/env.h:158:49: error: unknown type name ‘ulong’; did you mean ‘long’?
 ulong env_get_ulong(const char *name, int base, ulong default_val);

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
Tested-by: Joris Offouga <offougajoris@gmail.com>
Tested-by: Heiko Schocher <hs@denx.de>
---
v1 -> v2:
 - None

 include/env.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/env.h b/include/env.h
index a74a261..b72239f 100644
--- a/include/env.h
+++ b/include/env.h
@@ -9,6 +9,7 @@
 #ifndef __ENV_H
 #define __ENV_H
 
+#include <compiler.h>
 #include <stdbool.h>
 #include <linux/types.h>
 

From patchwork Mon Aug 26 11:06:18 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Pierre-Jean Texier <pjtexier@koncepto.io>
X-Patchwork-Id: 1153104
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=none (mailfrom) smtp.mailfrom=lists.denx.de
 (client-ip=81.169.180.215; helo=lists.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org; dmarc=fail (p=quarantine dis=none)
 header.from=koncepto.io
Authentication-Results: ozlabs.org;
 dkim=fail reason="signature verification failed" (2048-bit key;
 unprotected) header.d=koncepto.io header.i=@koncepto.io
 header.b="HvNQuMmn"; dkim-atps=neutral
Received: from lists.denx.de (dione.denx.de [81.169.180.215])
 by ozlabs.org (Postfix) with ESMTP id 46H8P275Mcz9sBF
 for <incoming@patchwork.ozlabs.org>;
 Mon, 26 Aug 2019 21:07:50 +1000 (AEST)
Received: by lists.denx.de (Postfix, from userid 105)
 id A5445C22143; Mon, 26 Aug 2019 11:07:11 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on lists.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=SPF_HELO_PASS, T_DKIM_INVALID
 autolearn=unavailable autolearn_force=no version=3.4.0
Received: from lists.denx.de (localhost [IPv6:::1])
 by lists.denx.de (Postfix) with ESMTP id 99A7EC21F9E;
 Mon, 26 Aug 2019 11:06:56 +0000 (UTC)
Received: by lists.denx.de (Postfix, from userid 105)
 id 1B211C21FB7; Mon, 26 Aug 2019 11:06:50 +0000 (UTC)
Received: from koncepto.io (koncepto.io [195.154.119.111])
 by lists.denx.de (Postfix) with ESMTPS id 5DD4FC22142
 for <u-boot@lists.denx.de>; Mon, 26 Aug 2019 11:06:47 +0000 (UTC)
Received: from menoah.home (lfbn-1-8923-205.w193-250.abo.wanadoo.fr
 [193.250.65.205])
 by koncepto.io (Postfix) with ESMTPSA id C256360379;
 Mon, 26 Aug 2019 13:06:46 +0200 (CEST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=koncepto.io;
 s=default; t=1566817607;
 bh=uZNRBSN4VpoP7mepjwChysqtnjGfdVboiTXTDe06LEU=;
 h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
 b=HvNQuMmnrlCD9XJG0/ozSpYPRFnOXBTD8TUFjcv0J+z/nZt0lCcYuSgICWNwLzjkK
 mOfgLxHOC5GAWv+jSVCNiL4NHnlsHtCWPTmwoWLHvuJMSopMyK/zbGkNKdiB0D5bAn
 oc3cAuzUIex4yqSidWtEXN7uxYstmMRNOVyyacTKDCfTYwUDb+4RMbr62MZGXNQO26
 0YShoLiU0geXqsu5IXy896mMWQejlZ6J14FiRN7ZD/cEHZGuPcFcophp4PgIMLnvJz
 KW2AAdMo8YoYUznZ19vBEmJuEutoju0h0Bee+YH0/jtbTgH9XarZrO6ljyEV+gbrKt
 W2vCG78kPreOQ==
From: Pierre-Jean Texier <pjtexier@koncepto.io>
To: u-boot@lists.denx.de
Date: Mon, 26 Aug 2019 13:06:18 +0200
Message-Id: <1566817578-16750-5-git-send-email-pjtexier@koncepto.io>
X-Mailer: git-send-email 2.7.4
In-Reply-To: <1566817578-16750-1-git-send-email-pjtexier@koncepto.io>
References: <1566817578-16750-1-git-send-email-pjtexier@koncepto.io>
Cc: bjorn@haxx.se, michal.simek@xilinx.com, joe.hershberger@ni.com,
 philip@philipmolloy.com
Subject: [U-Boot] [PATCH v2 4/4] ci: add envtools support
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.18
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <http://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>

This commit add envtools suppport to CI to verify if there
is no build issues.

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
Acked-by: Heiko Schocher <hs@denx.de>
---
v1 -> v2:
 - None

 .gitlab-ci.yml | 7 +++++++
 .travis.yml    | 4 ++++
 2 files changed, 11 insertions(+)

diff --git a/.gitlab-ci.yml b/.gitlab-ci.yml
index 84e79bf..a1c5b4f 100644
--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -145,6 +145,13 @@ Build tools-only:
   script:
     - make tools-only_config tools-only -j$(nproc)
 
+# Ensure env tools build
+Build envtools:
+  tags: [ 'all' ]
+  stage: testsuites
+  script:
+    - make tools-only_config envtools -j$(nproc)
+
 Run binman, buildman, dtoc and patman testsuites:
   tags: [ 'all' ]
   stage: testsuites
diff --git a/.travis.yml b/.travis.yml
index d330dda..6adc754 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -361,6 +361,10 @@ matrix:
     - name: "Build tools-only"
       script:
         - make tools-only_config tools-only -j$(nproc)
+    # Ensure env tools build
+    - name: "Build envtools"
+      script:
+        - make tools-only_config envtools -j$(nproc)
 
     # test/py
     - name: "test/py sandbox"
